tests/test_special_euclidean_group.py

```diff
@@ -1051,102 +1092,94 @@ def test_log_then_exp_right_with_angles_close_to_pi(self):
         """
         angle_types = self.angles_close_to_pi
         # Canonical inner product on the lie algebra
-        metric = self.metrics['right_canonical']
-        for base_point in self.elements.values():
-            for element_type in angle_types:
-                point = self.elements[element_type]
-                result = helper.log_then_exp(
-                                            metric=metric,
-                                            point=point,
-                                            base_point=base_point)
-
-                expected = self.group.regularize(point)
-
-                inv_rot_expected = np.zeros_like(expected)
-                inv_rot_expected[:, :3] = - expected[:, :3]
-                inv_rot_expected[:, 3:6] = expected[:, 3:6]
-                norm = np.linalg.norm(expected)
-                atol = RTOL
-                if norm != 0:
-                    atol = RTOL * norm
-
-                self.assertTrue(np.allclose(result, expected, atol=atol)
-                                or np.allclose(result, inv_rot_expected,
-                                               atol=atol),
-                                '\npoint = {}'
-                                '\nresult = {}'
-                                '\nexpected = {}'
-                                '\nexpected with opp rotation = {}'.format(
-                                   point,
-                                   result,
-                                   expected,
-                                   inv_rot_expected))
+        for metric in [self.metrics['right_canonical'],
+                       self.metrics['right_diag']]:
+            for base_point in self.elements.values():
+                for element_type in angle_types:
```

Comment:
That’s a lot of for loops... I’m usually less picky in testing code but eh... maybe  a local function definition to help make the thing less nested ?